app/controllers/api/account.php

```diff
@@ -703,6 +703,10 @@
         $loginSecret = Auth::tokenGenerator();
         $expire = DateTime::addSeconds(new \DateTime(), Auth::TOKEN_EXPIRATION_CONFIRM);
 
+        $events->setContext('token', new Document([
+            'secret' => $loginSecret,
+        ]));
```

Comment:
The unhashed secret is required in the event, so pass it to the handler using context.